vxblastsms/ussd.py

```diff
@@ -89,83 +56,98 @@ def handle_raw_inbound_message(self, message_id, request):
             )
             return
 
-        from_addr = values['msisdn']
-        provider = self.normalise_provider(values['provider'])
-        ussd_session_id = optional_values['ussdSessionId']
-
-        if optional_values['to_addr'] is not None:
+        if values['type'] == '2':  # resume session
             session_event = TransportUserMessage.SESSION_RESUME
-            to_addr = optional_values['to_addr']
-            content = optional_values['request']
+        elif values['type'] == '1':  # new session
+            session_event = TransportUserMessage.SESSION_NEW
         else:
+            # TODO: handle other types if necessary
+            # Default to new session for 3 & 4 for now
```

Comment:
Can you open an issue for handling 3 and 4 gracefully? Timeout shouldn't create a new session.